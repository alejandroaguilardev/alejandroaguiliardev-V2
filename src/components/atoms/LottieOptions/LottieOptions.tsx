import { useLottie } from "lottie-react";

interface IDefault {
    data: object;
    loop?: boolean;
    autoplay?: boolean;
}
export const LottieOptions = ({
    data,
    loop = true,
    autoplay = true,
}: IDefault) => {
    const options = {
        animationData: data,
        loop: loop,
        autoplay: autoplay,
    };

    const { View } = useLottie(options);

    return <>{View}</>;
};
