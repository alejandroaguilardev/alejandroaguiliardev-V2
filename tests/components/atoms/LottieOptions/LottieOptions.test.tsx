import { render, screen } from "@testing-library/react";
import { LottieOptions } from "../../../../src/components/atoms/LottieOptions/LottieOptions";



describe("LottieOptions", () => {
	test("toMacthSnapShot", () => {
		const { container } = render(
			<LottieOptions data={{}} loop={true} autoplay={false} />
		);
		expect(container).toMatchSnapshot();
		expect(screen.getByText("Lottie")).toBeTruthy();
	});
});
