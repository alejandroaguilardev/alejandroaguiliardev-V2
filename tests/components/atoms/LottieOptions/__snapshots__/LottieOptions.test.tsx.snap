// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LottieOptions toMacthSnapShot 1`] = `
<div>
  <h1>
    Lottie
  </h1>
</div>
`;
